### PR TITLE
Support world crawling and pass on errors

### PR DESCRIPTION
**Pull request by SinTh0r4s**
Hey, I really like what you did here. Super light weight! I used a fork in another project of mine. Are you interested in all or some of these changes?

Changes:
 - Exceptions are passed on to the user
 - Support world crawl and not just "give me data from chunk X/Z"
 - Some minor things
 - Kicked out main() entry point for tests. Minecrafts Forge Mod Loader does not like references to System.exit()

Happy to roll back some changed if you like

### CODE REVIEW

File: src/main/java/io/xol/enklume/nbt/NBTLong.java [hunk 1]
```diff
@@ -10,13 +10,13 @@
...
         super.feed(is);
 
-        data = is.read() << 56;
-        data += is.read() << 48;
-        data += is.read() << 40;
-        data += is.read() << 32;
-        data += is.read() << 24;
-        data += is.read() << 16;
-        data += is.read() << 8;
+        data = (long) is.read() << 56;
```
Comment (Hugobros3) [inline src/main/java/io/xol/enklume/nbt/NBTLong.java:13]: Was this broken then ? I guess I might have never realized/tested it

File: src/main/java/io/xol/enklume/MinecraftRegion.java [hunk 2]
```diff
@@ -103,12 +88,4 @@ private MinecraftChunk getChunkInternal(int x, int z) {
         }
         return new MinecraftChunk(x, z);
     }
-
-    public void close() {
```
Comment (Hugobros3) [inline src/main/java/io/xol/enklume/MinecraftRegion.java:107]: Not closing files isn't amazing, then again maybe there is already a finalizer taking care of this ? I don't know the fine print here

**Comment by Hugobros3** [comment]
Hey thanks that's pretty cool. I would merge this, just the readme changes would need to be left out.

A bit weird to me you'd need a library to parse Anvil in a Minecraft mod, doesn't forge just, you know, expose the official implementation ?

File: src/main/java/io/xol/enklume/nbt/NBTLong.java (hunk 1 above)
Comment (SinTh0r4s) [reply src/main/java/io/xol/enklume/nbt/NBTLong.java:13]: is.read() provides an integer. This is then shifted out of scope so you always have zero as result. Haven't testet is myself though...

**Comment by SinTh0r4s** [comment]
Reverted the readme. If all is included i can take the original as dependency anyway.

I started with a separate script-like program to regenerate some lost meta info about the world. It was supposed to run once and afterwards the new meta infos are saved. Now i integrated it into the mod. But i still guess, that your lib is significantly faster to crawl through the whole world. Especially on older servers. I just need the tile entity storage and Minecraft loads whole chunks including height maps, light data, .... -> slow

Again i really like this lib

**Comment by Hugobros3** [comment]
Looks okay. There is very little to this library ultimately, just a straightforward (and very incomplete) transcription of the format as described in various wikis, but I'm glad it helped someone!

File: src/main/java/io/xol/enklume/nbt/NBTLong.java (hunk 1 above)
Comment (Hugobros3) [reply src/main/java/io/xol/enklume/nbt/NBTLong.java:13]: I guess it depends how the type inference/checking works. I could see it working in a language if it starts with the knowledge the rhs to += must be a long, and then as you recurse into it keeps trying to have a long.

However, in Java this was clearly a bug: https://godbolt.org/z/fM5Wfe9e8

**Pull request merged**